### PR TITLE
Stubbed provided block parseing

### PR DESCRIPTION
**Pull request by BrianEdwards85**
Updated to parser to eat provided block, actual parsing/decomposition of provides to follow

### CODE REVIEW

**Comment by BrianEdwards85** [comment]
Yes, tests and actual parsing of provided will come next just wanted to get the consuming of provided done so I can get out of the way and prevent merge conflicts

File: src/smidje/parser.clj [hunk 1]
```diff
@@ -3,16 +3,36 @@
...
+(def provided "provided")
+
+(defn- provided-form?
+  [form]
+  (and (seq? form)
+       (= (first form) 'provided)))
+
+(defn- has-provided-form? [input]
+  (and (> (count input) 3)
+       (provided-form? (nth input 3))))
```
Comment (dustinvogt) [inline src/smidje/parser.clj]: In the future we'll need to expand this to: checking for a lhs, an arrow, a rhs, and metadata after the rhs (like :times 1)

File: src/smidje/parser.clj [hunk 2]
```diff
@@ -27,7 +47,7 @@
     ; TODO: error messages on bad syntax
     (if (and (> (count input) 2)
              (is-arrow (second input)))
-      (recur (conj result (parse-equals input)) (drop 3 input))
+      (recur (conj result (parse-equals input)) (drop (if (has-provided-form? input) 4 3 ) input))
```
Comment (dustinvogt) [inline src/smidje/parser.clj]: Does this allow us to put provided before or within the asserts? provided has to happen after all assertions

Comment (BrianEdwards85) [inline src/smidje/parser.clj]: My understanding of provided is that the provided block must immediately follow an assert, and only applies to the assert it follows

Comment (dustinvogt) [inline src/smidje/parser.clj]: Right, I'm completely wrong about this

**Pull request merged**